src/lib/helpers.ts

```diff
@@ -7,6 +7,20 @@ import Log from "./logger";
 import { Binding, CapabilityExport } from "./types";
 import { sanitizeResourceName } from "../sdk/sdk";
 
+export function matchesRegex(pattern: RegExp, testString: string): boolean {
+  // edge-case
+  if (!pattern) {
+    return false;
+  }
+
+  try {
+    const regex = new RegExp(pattern);
+    return regex.test(testString);
+  } catch {
+    return false;
+  }
```

Comment:
```suggestion
  try {
    const regex = new RegExp(pattern);
    return regex.test(testString);
  } catch {
    return false;
  }
```

Pretty sure this try catch is unnecessary and brings down coverage due to the fact that you can only add Regex to `WithRegexName` and `InRegexNamespace`, the error will never get to this function.